### ISSUE TITLE
Simple implementation of control naming of namespaces

### ISSUE THREAD

**Issue by gabrilabs75** [clastix/capsule#34]
# Describe the feature

We may need a simple implementation of control naming of namespaces.
Currently during creation of a new namespace by tenant-owner we receive an error message from k8s if the name of our selected namespace is already used by another tenant.
 
For example:

```
[ansible@k8s-caas-ansible-installer ~/capsule/hack]$ kubectl create ns production
Error from server (AlreadyExists): namespaces "production" already exists
```

# What would the new user story look like?

1. Tenant owner creates a new _Namespace_ (eg. production like the example above)

2. Capsule creates the new namespace having the name "tenantname-production" where the "tenantname" is the name of the tenant and "production" is the name of the namespace.

3. This is going to be attached to the _Tenant_

4. All the magic happens in the background

**Comment by bsctl** [comment]
@gabrilabs75 thanks for submitting this request. A control on the naming convention may have sense since it is likely that different tenant owners want to call their namespaces "production", "test", and "development".

Another viable solution is just instruct Capsule to deny the namespace creation if the name is not in the form of "tenantname"-"namespace" leaving the tenant owner to implement the naming convention himself.

Also I think this should be a configurable option in Capsule since this policy may be useful in some environments and too much annoying in other environments.

**Comment by prometherion** [comment]
> Another viable solution is just instruct Capsule to deny the namespace creation if the name is not in the form of "tenantname"-"namespace" leaving the tenant owner to implement the naming convention himself.

I guess this should be the preferred implementation, maybe providing a simple boolean flag to the CLI as `--force-tenant-prefix`.

tl;dr; with `--force-tenant-prefix` as _true_ the validation webhook will ensure the desired pattern, otherwise, it will reject the request.

WDYT @gabrilabs75?

**Comment by gabrilabs75** [comment]
@prometherion  it seems to be perfect for our scope...
Thanx in advance!
